test/Elastic.Apm.Tests/ApiTests/ApiTests.cs

```diff
@@ -358,5 +378,97 @@ public void TagsOnTransactionAndSpan()
 			payloadSender.SpansOnFirstTransaction[0].Tags.Should().Contain("fooSpan2", "barSpan2");
 			payloadSender.SpansOnFirstTransaction[0].Context.Tags.Should().Contain("fooSpan2", "barSpan2");
 		}
+
+		/// <summary>
+		/// Creates a transaction and then a span inside this transaction and then a 2. span within the 1. span (aka sub span)
+		/// Makes sure the relationship between the transaction and the spans are captured correctly.
+		/// </summary>
+		[Fact]
+		public void CreateSubSpan()
+		{
+			var payloadSender = new MockPayloadSender();
+			StartTransactionAndSpanWithSubSpan(payloadSender, s => { });
+		}
+
+		/// <summary>
+		/// Helper method. It creates a transaction, and a span on it, and then a sub span on that span.
+		/// Then it runs <paramref name="action" /> by passing the subspan as parameter
+		/// </summary>
+		/// <param name="payloadSender"></param>
+		/// <param name="action"></param>
+		private void StartTransactionAndSpanWithSubSpan(MockPayloadSender payloadSender, Action<ISpan> action)
+		{
+			using (var agent = new ApmAgent(new TestAgentComponents(payloadSender: payloadSender)))
+			{
+				var transaction = agent.Tracer.StartTransaction(TestTransaction, UnitTest);
+				Thread.Sleep(5);
```

Comment:
Could you please clarify what is the purpose of this `Thread.Sleep` call?

Reply:
I just thought that it's more realistic than spans with 0 duration.